storybookjs/storybook
# TestBuild: Globalize `@storybook/blocks` if `build.test.emptyBlocks` is `true`

## What I did

The recently introduced `--test` flag currently doesn't toggle much functionality.

This introduces the first change in functionality based on this flag:
- When the flag is true an `build.test.emptyBlocks` becomes true (by default, can be switched back to false in `main.ts`).
- This causes the preview builders to globalize `@storybook/blocks`.
- The effect of that is that the builder can skip those imports, thus making the bundle-process faster and the generated `storybook-static`'s size smaller.
- It will cause actual usage of `@storybook/blocks` to instantly fail.

The idea is that `--test` will gain more feature flags like these that eliminate docs completely.
This is just 1 small step in that direction.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

- Create a sandbox
- Run `sb build --test` in the sandbox
- Check if `@storybook/blocks` exists/is globalized

Do the above for both a webpack based and vite based sandbox.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
